### NUMPY ARRAYS

ARRAYS VS LIST COMPARISON

In [1]:
import numpy as np

In [2]:
#list
lst=[1,2,3,4]
print(lst)
print(type(lst))

[1, 2, 3, 4]
<class 'list'>


In [3]:
#numpy array
arr=np.array([1,2,3,4])
print(arr)
print(type(arr))

[1 2 3 4]
<class 'numpy.ndarray'>


In [4]:
print(arr==lst)
print(arr.tolist()==lst)

[ True  True  True  True]
True


ADDING THE ELEMENTS TO THE LIST AND NUMPY ARRAY

In [5]:
#appending in the list
lst=[1,2,3]
print(lst)
lst.append(4)
print(lst)

[1, 2, 3]
[1, 2, 3, 4]


In [6]:
#appending in numpy
arr2=np.array([1,2,3])
print(arr2)
arr2=np.append(arr2,4)
print(arr2)

[1 2 3]
[1 2 3 4]


DELETING IN NUMPY ARRAY VS LSITS

In [7]:
#deleting an element in a list
lst=[1,2,3,4]
print(lst)
lst.pop(3)
print(lst)

[1, 2, 3, 4]
[1, 2, 3]


In [13]:
#Deleting an element in a numpy
arr3=np.array=(1,2,3,4)
print(arr3)
arr3=np.delete(arr3,-1)
print(arr3)

(1, 2, 3, 4)
[1 2 3]


UPDATING ELEMENTS IN AN ARRAY AND A LIST

In [9]:
lst=[1,2,3,4]
print(lst)
lst[-1]=5
print(lst)


[1, 2, 3, 4]
[1, 2, 3, 5]


In [18]:
arr4=np.array=([1,2,3,4])
print(arr4)
arr4[-1]=5
print(arr4)

[1, 2, 3, 4]
[1, 2, 3, 5]


ARRAYS VS LIST COMPARISON(in terms of time)

computing time taken by list

In [34]:
import time
my_list=list(range(10000000))
my_arr=np.arange(10000000)

print(len(my_list))
print(len(my_arr))

start_time=time.time()

for i in range(100):
    if 5000000 in my_list:
        pass

end_time=time.time()

print("time taken by list to execute is :",end_time-start_time)


10000000
10000000
time taken by list to execute is : 5.532519817352295


computing time taken by array

In [33]:
import time

my_arr=np.arange(10000000)

start_time=time.time()

for i in range(100):
    if 5000000 in my_arr:
        pass

end_time=time.time()

print("time taken by array to execute is :",end_time-start_time)



time taken by array to execute is : 0.8527474403381348


NOTE:

time taken by list to execute is : 5.532519817352295

time taken by array to execute is : 0.8527474403381348

ARRAY is five times faster than the list